yesod-core/src/Yesod/Core/Class/Yesod.hs

```diff
@@ -249,6 +249,17 @@ class RenderRoute site => Yesod site where
     jsAttributes :: site -> [(Text, Text)]
     jsAttributes _ = []
 
+    -- | Same as @jsAttributes@ but allows you to run arbitrary Handler code
+    --
+    -- This is useful if you need to add a randomised nonce value to the script
+    -- tag generated by @widgetFile@. If you use both this function and
+    -- @jsAttributes@, the two attribute maps will be merged, favouring the
+    -- keys from @jsAttributes@ in case of a collision.
+    --
+    -- @since 1.6.16
+    jsAttributesHandler :: HandlerFor site [(Text, Text)]
```

Comment:
The way we've handled this for other. similar cases is to have a default implementation of `jsAttributesHandler` which uses `jsAttributes`. Then we document that if `jsAttributesHandler` is overridden, `jsAttributes` is ignored.

Reply:
Ok, I think that is the better approach. I'll make that change.